documentation/userguide.md

```diff
@@ -0,0 +1,270 @@
+# Teamscale .NET Profiler
+
+Test Gap analysis is a tool, which brings more transparency to the testing process. To this end, changes within an application are identified and matched against information about test execution. From a technical perspective, this requires two types of information: 
+
+* the source code at the current revision
+* execution information, which is collected via the .NET Profiler
+
+The profiler writes method coverage information of a .NET application into a report file (a.k.a. *trace file*). It uses the standard .NET profiling interface and therefore can be used with any .NET application.
+
+The trace file is created immediately after the first call of a .NET method. However, it remains mostly empty as long as the process runs. This avoids an unnecessary performance overhead due to file accesses. As soon as the process ends gracefully, the report is finished and all collected information is written to the trace file.
+
+In the case of long running processes it must be taken into account that the report file is written only after the process has ended. In such a case, it must be ensured that the application is stopped and restarted in regular intervals or eager writing of the trace file is enabled.
```

Comment:
This completely ignores the "eagerness" mode. Update?

Reply:
The last part of the sentence is about eager mode